### PR TITLE
[WIP] Pull the image from private registry

### LINKED ISSUE #2537: Test script requires authentication from registry.redhat.io
[kind/Feature]

## Which functionality do you think we should add?

Actually our test script runs inside a container (https://github.com/openshift/release/blob/master/ci-operator/config/openshift/odo/openshift-odo-master__v4.1.yaml#L10). So for this container, may be I need to get the secret and apply/use that to the running cluster on aws.

To pull the supported image available only in `registry.redhat.io` we need account service credentials to be configured in the main CI and should be mounted to the test container.

## Why is this needed?

To fix the issue where account service credentials are required. eg: Pulling supported builder images from `registry.redhat.io`

### PR DESCRIPTION
**Pull request by prietyc123**
**What type of PR is this?**

/kind bug

**What does does this PR do / why we need it**:

It pulls the images from private registry.

**Which issue(s) this PR fixes**:

Fixes #2537 

**How to test changes / Special notes to the reviewer**:
```$ make test-e2e-images```

### CODE REVIEW

File: scripts/configure-installer-tests-cluster.sh [hunk 1]
```diff
@@ -9,11 +9,34 @@ HTPASSWD_SECRET="htpasswd-secret"
...
+REGISTRY_UN=""
+REGISTRY_PASS=""
+
 # Default values
 OC_STABLE_LOGIN="false"
 # Exported to current env
 export KUBECONFIG=${KUBECONFIG:-"${DEFAULT_INSTALLER_ASSETS_DIR}/auth/kubeconfig"}
 
+# Mount path of odo secret directory
+oc get secret
```
Comment (amitkrout) [inline scripts/configure-installer-tests-cluster.sh]: @prietyc123 can you check it with ```oc get secret -n odo```

**Comment by prietyc123** [comment]
```
error: could not run steps: step [release:latest] failed: no 'cli' image was tagged into the stable stream, that image is required for building a release
```

/retest

**Comment by prietyc123** [comment]
I am not able to find any secret in - https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_odo/2668/pull-ci-openshift-odo-master-v4.1-integration-e2e-benchmark/1021#1:build-log.txt%3A44

@amitkrout Can please check with upstream team why there is no secret file in the mount path even after mounting it in the path - https://github.com/openshift/release/blob/master/ci-operator/config/openshift/odo/openshift-odo-master__v4.1.yaml#L21-L22

/hold

**Comment by amitkrout** [comment]
> I am not able to find any secret in - https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_odo/2668/pull-ci-openshift-odo-master-v4.1-integration-e2e-benchmark/1021#1:build-log.txt%3A44
> 
> @amitkrout Can please check with upstream team why there is no secret file in the mount path even after mounting it in the path - https://github.com/openshift/release/blob/master/ci-operator/config/openshift/odo/openshift-odo-master__v4.1.yaml#L21-L22
> 
> /hold

Sure, will check with upstream CI team. Thanks

**Comment by prietyc123** [comment]
> > @amitkrout Can please check with upstream team why there is no secret file in the mount path even after mounting it in the path - https://github.com/openshift/release/blob/master/ci-operator/config/openshift/odo/openshift-odo-master__v4.1.yaml#L21-L22
> 
> @prietyc123 Probably you need to create a dir of `/tmp/sccret` explicitly in the [container](https://github.com/openshift/odo/blob/master/openshift-ci/build-root/Dockerfile) itself where our odo test launches.

Done

**Comment by prietyc123** [comment]
> > > @amitkrout Can please check with upstream team why there is no secret file in the mount path even after mounting it in the path - https://github.com/openshift/release/blob/master/ci-operator/config/openshift/odo/openshift-odo-master__v4.1.yaml#L21-L22
> > 
> > 
> > @prietyc123 Probably you need to create a dir of `/tmp/sccret` explicitly in the [container](https://github.com/openshift/odo/blob/master/openshift-ci/build-root/Dockerfile) itself where our odo test launches.

@amitkrout @mohammedzee1000 I have created dir ```/tmp/secret``` but there is no secret created in the dir. - https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_odo/2668/pull-ci-openshift-odo-master-v4.1-integration-e2e-benchmark/1159#1:build-log.txt%3A151 . Can you please check once?

**Comment by prietyc123** [comment]
/hold

https://github.com/openshift/odo/pull/2668#issuecomment-599364206

**Comment by amitkrout** [comment]
> > > @amitkrout Can please check with upstream team why there is no secret file in the mount path even after mounting it in the path - https://github.com/openshift/release/blob/master/ci-operator/config/openshift/odo/openshift-odo-master__v4.1.yaml#L21-L22
> 
> > @prietyc123 Probably you need to create a dir of `/tmp/sccret` explicitly in the [container](https://github.com/openshift/odo/blob/master/openshift-ci/build-root/Dockerfile) itself where our odo test launches.
> 
> @amitkrout @mohammedzee1000 I have created dir `/tmp/secret` but there is no secret created in the dir. - https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_odo/2668/pull-ci-openshift-odo-master-v4.1-integration-e2e-benchmark/1159#1:build-log.txt%3A151 . Can you please check once?

@prietyc123 thanks for the quick analysis. I did follow all the steps in the pr https://github.com/openshift/release/pull/6764 to make this work. I am not sure why it is not working. The quicker solution i ma seeing atm is lets arrange a bjn call with @hongkailiu. I will create a calendar invite after discussing the availability of @hongkailiu.

**Comment by prietyc123** [comment]
> > > > @amitkrout Can please check with upstream team why there is no secret file in the mount path even after mounting it in the path - https://github.com/openshift/release/blob/master/ci-operator/config/openshift/odo/openshift-odo-master__v4.1.yaml#L21-L22
> > 
> > 
> > > @prietyc123 Probably you need to create a dir of `/tmp/sccret` explicitly in the [container](https://github.com/openshift/odo/blob/master/openshift-ci/build-root/Dockerfile) itself where our odo test launches.
> > 
> > 
> > @amitkrout @mohammedzee1000 I have created dir `/tmp/secret` but there is no secret created in the dir. - https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/openshift_odo/2668/pull-ci-openshift-odo-master-v4.1-integration-e2e-benchmark/1159#1:build-log.txt%3A151 . Can you please check once?
> 
> @prietyc123 thanks for the quick analysis. I did follow all the steps in the pr [openshift/release#6764](https://github.com/openshift/release/pull/6764) to make this work. I am not sure why it is not working. The quicker solution i ma seeing atm is lets arrange a bjn call with @hongkailiu. I will create a calendar invite after discussing the availability of @hongkailiu.

Sure 👍

File: scripts/configure-installer-tests-cluster.sh [hunk 2]
```diff
@@ -11,12 +11,35 @@ SETUP_OPERATORS="./scripts/setup-operators.sh"
 DEFAULT_INSTALLER_ASSETS_DIR=${DEFAULT_INSTALLER_ASSETS_DIR:-$(pwd)}
 KUBEADMIN_USER=${KUBEADMIN_USER:-"kubeadmin"}
 KUBEADMIN_PASSWORD_FILE=${KUBEADMIN_PASSWORD_FILE:-"${DEFAULT_INSTALLER_ASSETS_DIR}/auth/kubeadmin-password"}
+
+# Registry.redhat.io username and password for local testing
+REGISTRY_UN=""
+REGISTRY_PASS=""
+
```
Comment (amitkrout) [inline scripts/configure-installer-tests-cluster.sh]: Can you please execute just ```env``` in this script to check what are the environment variable available/injected in the test container.

**Comment by prietyc123** [comment]
> @prietyc123 This is the input we got form testplatform team in the regard of secret mount.
> 
> ```
> @arout @Priti so... I'm really sorry about this, but I just consulted this with team members who have
>  a better idea about this area, and the current state is that the custom secrets passed to test 
> workloads like this cannot (at the moment) be accessed by the template-based test workloads, but 
> only by container-based tests (=not e2e tests)
> ```
> 
> Till now they have not add the support for what we are looking here, i mean container-based e2e tests. The good news is that @stevekuznetsov has created ticket to add support for our requirement and he will start working on the ticket in their vert next sprint.
> 
> @prietyc123 i think there is no point to keep this pr open as fix from OpenShift CI side may take more than one sprint. BTW its your call whether to keep it or close it atm :)

I agreed to your point @amitkrout. As per our discussion with test platform team I think its better we close this pr for now because it may take more than one sprint for them to add it.

**Comment by prietyc123** [comment]
Closing this pr as test platform team is not having support for e2e test till now. More details : https://github.com/openshift/odo/pull/2668#pullrequestreview-378918703

**Pull request closed without merging**